body.tex

```diff
@@ -1,6 +1,136 @@
+
+Apache Kafka is a distributed data streaming platform that is designed for high throughput and fault tolerance. Sasquatch uses the Strimzi operator\footnote{\url{https://strimzi.io}} to deploy Apache Kafka on Kubernetes. Kafka is used to capture telemetry data before writing to InfluxDB. For fault tolerance, Kafka replicates data across multiple brokers in the cluster. Each metric or telemetry topic is configured with a replication factor of three and data durability is ensured by requiring acknowledgment from at least two replicas when writing to Kafka.
+
+Figure 1 shows the Sasquatch deployment at USDF. MirrorMaker2 is the component responsible for mirroring the data from the Summit \cite{SQR-050}. The data is then persisted in the Kafka brokers with a retention period of 72h. Kafka Connect is the component responsible for running the connectors which consume data from the Kafka brokers write to InfluxDB.
+
+\begin{figure}[h]
+    \centering
+    \includegraphics[width=0.8\textwidth]{figures/sasquatch-architecture.pdf}
+    \caption{Sasquatch deployment at the US Data Facility.}
+    \label{fig:system-architecture}
+
+\end{figure}
+
+Sasquatch implements namespaces to organize and identify different data sources. For example, the namespace \texttt{lsst.sal} is used by Sasquatch to identify Kafka topics produced by the OCS. Kafka Connect then route the data to the corresponding database in InfluxDB, in this example, the Engineering and Facility Database (EFD). Similarly, other namespaces are used to identify metrics data produced by the LSST Science Pipelines and route the data to the corresponding databases in InfluxDB.
+
+\subsection{Sending data to Sasquatch}
+
+Sasquatch uses Avro for the data serialization format and the Confluent Schema Registry (SR)\footnote{\url{https://docs.confluent.io/platform/current/schema-registry/}} to store the Avro schemas and ensure they can evolve safely, whitout breaking the Sasquatch consumers.
+
+Applications connecting to Sasquatch can use native Kafka and SR clients to send data to Kafka. For example, the Kafka producer\footnote{\url{https://ts-salkafka.lsst.io}} that consumes data from the OCS uses the \texttt{confluent-kafka-python} client to send data to Kafka and the Confluent SR client to register the Avro schemas. Avro schemas are created from the XML schemas in the OCS code base\footnote{\url{https://ts-xml.lsst.io}}.
+
+Alternatively the REST Proxy\footnote{\url{https://docs.confluent.io/platform/current/kafka-rest/index.html}} provides a REST interface to Kafka, making it easy to produce and consume messages. The LSST Analysis Tools package measure metrics from the LSST Pipelines outputs uses the REST Proxy to send data to Kafka.
```

Comment:
I feel like there's a verb missing from "package measure metrics"...maybe "package used to measure metrics" ?  Or "that measures metrics" ?